src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -134,6 +134,7 @@ private static bool IsTypeImplicitlyConvertible(Compilation compilation, ITypeSy
         {
             var symbolGroups = from symbol in symbols
```

Comment:
you might want to ensure that hte code that produces these symbols does a .CanBeReferencedByName on them.

Reply:
`CanBeReferencedByName` is actually true for the empty symbol. @333fred is this expected?

Reply:
Are these local symbols? If so, that is current expected behavior, but there's certainly an argument to be made that a local symbol with no name cannot be referenced by name :). The current behavior dates back to before roslyn went open source, so changing it is a risky proposition, but  I could see that as a valid change.

Reply:
Everything else checks: 

```
                return SyntaxFacts.IsValidIdentifier(this.Name) &&
                    !SyntaxFacts.ContainsDroppedIdentifierCharacters(this.Name);
```

This really seems like the case for `locals/labels/aliases/rangevariablse`

Reply:
Yes, this is a local. Not sure what the bar is for back-compat, but I can file a bug to track this

Reply:
IMO, this should be a totally fine change, and def within the bar we have for changing behavior of existing apis.

Reply:
Filed #47690